spec/controllers/home_controller_spec.rb

```diff
@@ -4,9 +4,60 @@
   render_views
 
   describe 'GET #index' do
-    it 'redirects to about page' do
-      get :index
-      expect(response).to redirect_to(about_path)
+    context 'when not signed in' do
+      it 'redirects to about page' do
+        get :index
+        expect(response).to redirect_to(about_path)
+      end
+    end
+
+    context 'when signed in' do
+      let(:user) { Fabricate(:user) }
+      subject do
+        sign_in(user)
+        get :index
+      end
+
+      it 'assigns @body_classes' do
+        subject
+        expect(assigns(:body_classes)).to eq 'app-body'
```

Comment:
> I think my preference would be that we eventually remove that gem, and improve either feature specs or request specs as is relevant in each case.

In my opinion, most things to be "improved" is in the implementation, not the spec.

> But it's worth asking yourself why your controller is so complicated that
sharing it without a view really makes sense. To me it would indicate that
you should be extracting logic into models or concerns.

https://github.com/rails/rails/issues/18950#issuecomment-114613479

If we do such extractions, it would make sense to remove `assigns`, etc. It is, however, out of scope for the series of pull requests. Using the gem is still legitimate for the situation.

By the way, if you are for extracting logic to concerns, why not open an issue and clarify the direction? I think it is better to have such important decisions earlier, whichever direction we go.

> Coverage is coverage, even if it covers a bit too much detail.

I believe that is not too much. The problem is where it should be tested; the interface between models/concerns and controllers, or one between controllers and views.